### PR TITLE
remove forgetting of controller keys in case it failed more than 5 times

### LINKED ISSUE #1851: Do not stop processing items in controllers after X failed retries
**User Story**

As a developer I want my controllers to try achieving the desired states forever with an ever-increasing delay. Currently, we have in various controllers some logic that stops after N retries, e.G.:

```
if ctrl.queue.NumRequeues(key) < 5 {
		ctrl.queue.AddRateLimited(key)
￼ 		return
￼ 	}
``` 

**Acceptance criteria**

* All controllers within kubermatic were checked if they have a retry-limiting logic and if yes, it was removed
* The machine-controller does not have a retry-limiting logic anymore

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Remove forgetting of controller keys in case it failed more than 5 times.
This will ensure that we'll always retry with a configured backoff - potentially reaching the maximum backoff.

Fixes #1851

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/controller/kubeletdnat/kubeletdnat_controller.go [hunk 1]
```diff
@@ -121,7 +104,21 @@ func (ctrl *Controller) processNextItem() bool {
 
 	defer ctrl.queue.Done(key)
 	err := ctrl.syncDnatRules()
-	ctrl.handleErr(err, key)
+	if err == nil {
+		// Forget about the #AddRateLimited history of the key on every successful synchronization.
+		// This ensures that future processing of updates for this key is not delayed because of
+		// an outdated error history.
+		ctrl.queue.Forget(key)
```
Comment (alvaroaleman) [inline api/pkg/controller/kubeletdnat/kubeletdnat_controller.go]: What if this key is currently in the queue?

Comment (mrIncompetent) [reply api/pkg/controller/kubeletdnat/kubeletdnat_controller.go]: Same bahavior as before I guess - which I'm not aware of

Comment (alvaroaleman) [reply api/pkg/controller/kubeletdnat/kubeletdnat_controller.go]: :D Can you check on it?

Comment (mrIncompetent) [reply api/pkg/controller/kubeletdnat/kubeletdnat_controller.go]: So checking the interface from the queue reveals that calling `Forget` will just reset the rate-limiter:
```
// This only clears the `rateLimiter`, you
// still have to call `Done` on the queue.
```

**Review by alvaroaleman — DISMISSED** [review]

**Review by alvaroaleman — APPROVED** [review]

**Review by p0lyn0mial — APPROVED** [review]

**Pull request merged**